navigation/navigation-runtime/src/webMain/kotlin/androidx/navigation/BrowserHistory.kt

```diff
@@ -89,7 +89,11 @@ internal suspend fun BrowserWindow.bindToNavigation(
                 if (entries.isEmpty()) return@collect
                 val routes = entries.map { it.getRouteWithArgs() ?: return@collect }
 
-                val newUri = appAddress + getBackStackEntryPath(entries.last())
+                val routeFragment = getBackStackEntryRoute(entries.last())
+                    .takeIf { it.isNotBlank() }
+                    ?.let { "#$it" }
```

Comment:
Enforcing putting everything into url fragment just for fixing one case doesn't seem valid. We can change default, but removing the ability seems wrong

Reply:
No. It is generally a more correct way. We must handle app routes locally and don't call a backend.
(Flutter navigation uses the same approach)